### PR TITLE
raise RuntimeError appropriately for animation update func

### PR DESCRIPTION
**Pull request by johnnyEmpires**
## PR Summary

Current `_draw_frame` method only checks if `_drawn_artists` is `None` and raise RuntimeError. There are other cases that needs to be checked but are not handled correctly.

* user forgot to return sequence of artist (returns None)
* user (for some reason) returned a string or other object type...AttributeError will be raised
* user returns a sequence of other objects e.g. a string instead of Artist
* user forgot to put comma to return a sequence, TypeError will be raised (same with returning a number or bool)

This can be confusing to new users of `FuncAnimation` since different errors are raised instead of RuntimeError `The animation function must return a sequence of Artist objects`.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in - README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
@dopplershift  can you take a look at this?  I think you are the animation guru.

@johnnyEmpires the description above is a bit vague.  What are the "other cases" that need to be checked?

File: lib/matplotlib/tests/test_animation.py [hunk 1]
```diff
@@ -272,3 +272,40 @@ def frames_generator():
...
         assert (f() is None) != cache_frame_data
+
+
+def test_draw_frame():
+    # test _draw_frame method
+
+    fig, ax = plt.subplots()
+    line, = ax.plot([])
+
+    def init():
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_animation.py:283]: `init` isn't used.

File: lib/matplotlib/tests/test_animation.py [hunk 2]
```diff
@@ -272,3 +272,40 @@ def frames_generator():
...
+        # user (for some reason) returned a string...AttributeError is raised
+        line.set_data([0, 1], [0, i])
+        return 'a string'
+
+    def animate_case_4(i):
+        # user returns a sequence other objects instead of Artist.
+        line.set_data([0, 1], [0, i])
+        return 'a string',
+
+    with pytest.raises(RuntimeError) as context:
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_animation.py:307]: `context` isn't used.

**Comment by johnnyEmpires** [comment]
Thanks @jklymak and @dopplershift for checking my PR. It's actually my first time to do it, thus the several commits as I try to understand why codecov test is failing (not still sure why)...I updated PR summary, could you check it?

I'll commit next the updated test codes.

**Pull request merged**